src/smart-components/portfolio/share-portfolio-modal.js

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import FormRenderer from '../common/form-renderer';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Modal } from '@patternfly/react-core';
+import { createPortfolioShareSchema } from '../../forms/portfolio-share-form.schema';
+import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
+import { fetchPortfolios } from '../../redux/actions/portfolio-actions';
+import { fetchShareInfo, sharePortfolio, unsharePortfolio } from '../../redux/actions/share-actions';
+import { fetchRbacGroups } from '../../redux/actions/rbac-actions';
+import { pipe } from 'rxjs';
+
+const SharePortfolioModal = ({
+  history: { goBack },
+  isLoading,
+  addNotification,
+  fetchPortfolios,
+  initialValues,
+  fetchShareInfo,
+  sharePortfolio,
+  fetchRbacGroups,
+  shareInfo,
+  portfolioId,
+  rbacGroups
+}) => {
+  useEffect(() => {
+    fetchShareInfo(portfolioId);
+    fetchRbacGroups();
+  }, []);
+
+  const initialShares = () => {
+    let initialGroupShareList = shareInfo.map((group) => { const groupName = group.group_name;
+      let options = permissionOptions.find(perm => (perm.value === group.permissions.sort().join(',')));
+      return { [groupName]: options ? options.value : 'Unknown' };});
+    let initialShareList = initialGroupShareList.reduce((acc, curr) => ({ ...acc, ...curr }), {});
+
+    return initialShareList;
+  };
+
+  const onSubmit = data =>
+  {
+    let sharePromises = [];
+    if (data.group_uuid && data.permissions) {
+      sharePromises.push(sharePortfolio(data));
+    }
+
+    shareInfo.map(share => {
+      let initialPerm = share.permissions.sort().join(',');
+      if (data[share.group_name] !== initialPerm) {
+        if (!data[share.group_name]) {
+          sharePromises.push(unsharePortfolio({ id: portfolioId, permissions: share.permissions, group_uuid: share.group_uuid }));
+        }
+        else {
+          if (share.permissions.length > data[share.group_name].split(',').length) {
+            sharePromises.push(unsharePortfolio({
+              id: portfolioId,
+              permissions: [ 'catalog:portfolios:write' ],
+              group_uuid: share.group_uuid
+            }));
+          }
+          else {
+            sharePromises.push(sharePortfolio({
+              id: portfolioId,
+              permissions: data[share.group_name],
+              group_uuid: share.group_uuid
+            }));
+          }
+        }
+      }
+    });
+
+    Promise.all(sharePromises).then(goBack).then(() => fetchPortfolios());
```

Comment:
I know that `goBack` is being used in some other modals, but i think it would be better to use `history.push(route)`. Go back might bite us when there is no route to go back to (when you open the page directly from URL for instance).

I plan to replace most of the `goBack` calls in the rest of app soon.

Reply:
Agree that we have to replace the goBack. I will leave this as is for now, as it is a modal and can be displayed from both the portfolios list a well as an individual portfolio detail page. We can replace it here when we decide on a common way to handle the back path for modals that can return to different pages ( parent)